src/Analysis/Codelyzer.Analysis.CSharp/CSharpRoslynProcessor.cs

```diff
@@ -247,6 +254,16 @@ public override UstNode VisitBlock(BlockSyntax node)
                 }
             }
 
+            var returnStatements = node.DescendantNodes().OfType<ReturnStatementSyntax>();
+            foreach (var returnStatement in returnStatements)
+            {
+                var r = VisitReturnStatement((ReturnStatementSyntax)returnStatement);
+                if (r != null)
+                {
+                    handler.UstNode.Children.Add(r);
+                }
+            }
+
```

Comment:
Did you check if we get any instances in the method declaration but outside the block/expressionbody? Same question applies to arrow statement.

Reply:
From what I've seen, return statements do not exist in method declarations while outside the block... If you do have an example of that, I'm all ears. One thing I did find though was that return statements are not allowed in Arrow Expressions (but they are permitted in Lambda Expressions which we don't support at the moment) so I will update my this PR to reflect that. 

As for Lambda Expression support, I can implement that as well, but I think it should be saved for another PR to prevent the scope from continuously incrementing (the purpose of this PR is to implement minimum functionality to support CoreWebApi and CoreMvc project detection). I'll open an issue for that and prioritize it next, but let me know if you'd like to discuss this further.